soroban-env-host/benches/common/measure.rs

```diff
@@ -285,11 +299,49 @@ mod cpu {
     }
 }
 
-fn measure_costs_inner<HCM: HostCostMeasurement, F>(
+fn harness<HCM: HostCostMeasurement, R>(
+    host: &Host,
+    mem_tracker: &MemTracker,
+    cpu_insn_counter: &mut cpu::InstructionCounter,
+    alloc_group_token: &mut AllocationGroupToken,
+    iterations: u64,
+    runner: &mut R,
+    samples: Vec<<<HCM as HostCostMeasurement>::Runner as CostRunner>::SampleType>,
+) -> Measurement
+where
+    R: FnMut(&Host, Vec<<<HCM as HostCostMeasurement>::Runner as CostRunner>::SampleType>),
+{
+    // TODO get rid of dependency of inputs on samples
+    let sample = samples[0].clone();
+    // start the cpu and mem measurement
+    let start = Instant::now();
+    mem_tracker.0.store(0, Ordering::SeqCst);
+    let alloc_guard = alloc_group_token.enter();
+    host.as_budget().reset_unlimited();
+    cpu_insn_counter.begin();
+    runner(host, samples);
+    // collect the metrics
+    let cpu_insns = cpu_insn_counter.end_and_count() / iterations;
+    drop(alloc_guard);
+    let stop = Instant::now();
```

Comment:
I think you can move the instant counter up above the drop of the alloc guard.

Reply:
Fixed.